packages/block-editor/src/components/inserter/index.native.js

```diff
@@ -101,23 +216,71 @@ class Inserter extends Component {
 export default compose( [
 	withSelect( ( select, { clientId, isAppender, rootClientId } ) => {
 		const {
-			getInserterItems,
 			getBlockRootClientId,
 			getBlockSelectionEnd,
+			getBlockOrder,
+			getBlockIndex,
+			getBlock,
 		} = select( 'core/block-editor' );
 
-		let destinationRootClientId = rootClientId;
-		if ( ! destinationRootClientId && ! clientId && ! isAppender ) {
-			const end = getBlockSelectionEnd();
-			if ( end ) {
-				destinationRootClientId = getBlockRootClientId( end ) || undefined;
+		const end = getBlockSelectionEnd();
+		const isAnyBlockSelected = ( ! isAppender && end );
+		const destinationRootClientId = isAnyBlockSelected ?
```

Comment:
Looks like a fix! Was it not inserting at the right place before that?

Reply:
Previously `withSelect` implementation of `Inserter` was returning `items` and it was not used at all. The changes are actually not related with each other.